source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -210,10 +210,11 @@ void TcpStatsdSink::TlsSink::commonFlush(const std::string& name, uint64_t value
   // This written this way for maximum perf since with a large number of stats and at a high flush
   // rate this can become expensive.
   const char* snapped_current = current_slice_mem_;
-  memcpy(current_slice_mem_, parent_.getPrefix().c_str(), parent_.getPrefix().size());
+  memcpy(current_slice_mem_, parent_.getPrefix().c_str(), // NOLINT(safe-memcpy)
```

Comment:
while you are in here can you make this change:
```
const std::string prefix = parent_.getPrefix();
memcpy(current_slice_mem_, prefix.data(), prefix.size()); // NOLINT(safe-memcpy)
 current_slice_mem_ += prefix.size();
```
There are 2 inefficiencies in the previous code that are easy to fix while you are in here:
1. Extra calls to getPrefix() may generates 3 copies of the string, which causes extra allocations and copies
1. Calling c_str() on a string that does not need to be null terminated is an anti-pattern. data() is preferred.